src/local_sensitivity/adjoint_common.jl

```diff
@@ -23,7 +27,25 @@ function adjointdiffcache(g,sensealg,discrete,sol,dg;quad=false)
   numparams = length(p)
   numindvar = length(u0)
   isautojacvec = get_jacvec(sensealg)
-  J = isautojacvec ? nothing : similar(u0, numindvar, numindvar)
+
+  issemiexplicitdae = false
+  mass_matrix = sol.prob.f.mass_matrix'
+  if mass_matrix isa UniformScaling
+    factorized_mass_matrix = mass_matrix
+  else
+    diffvar_idxs = findall(x->any(!iszero, @view(mass_matrix[:, x])), axes(mass_matrix, 2))
```

Comment:
why axes instead of eachcol?

Reply:
Because I need indices. `findall(x->any(!iszero, x), eachcol(M))` won't work.